forum/index.php

```diff
@@ -1,38 +1,41 @@
 <?php
 /*
-	Question2Answer by Gideon Greenspan and contributors
-	http://www.question2answer.org/
+    Question2Answer by Gideon Greenspan and contributors
+    http://www.question2answer.org/
 
-	File: index.php
-	Description: A stub that only sets up the Q2A root and includes qa-index.php
+    File: index.php
+    Description: A stub that only sets up the Q2A root and includes qa-index.php
 
 
-	This program is free software; you can redistribute it and/or
-	modify it under the terms of the GNU General Public License
-	as published by the Free Software Foundation; either version 2
-	of the License, or (at your option) any later version.
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License
+    as published by the Free Software Foundation; either version 2
+    of the License, or (at your option) any later version.
 
-	This program is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-	GNU General Public License for more details.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
 
-	More about this license: http://www.question2answer.org/license.php
+    More about this license: http://www.question2answer.org/license.php
 */
-//	Set base path here so this works with symbolic links for multiple installations
+
 require_once 'vendor/autoload.php';
 
+// Set base path here so this works with symbolic links for multiple installations
+define('QA_BASE_DIR', dirname(empty($_SERVER['SCRIPT_FILENAME']) ? __FILE__ : $_SERVER['SCRIPT_FILENAME']) . '/');
+
+if (file_exists(QA_BASE_DIR . 'qa-config.php')) {
+    require_once QA_BASE_DIR . 'qa-config.php';
+}
+
 global $client;
-$client = new Raven_Client('',['environment' => 'local']);
+$client = new Raven_Client(SENTRY_DSN ?? '', [
+    'environment' => SENTRY_ENVIRONMENT ?? 'local'

```

Comment:
bez sensu - przecież powinno brać bezpośrednio tylko i wyłącznie z SENTRY_ENVIRONMENT

Reply:
Z sensem, zapoznaj się z moim rozwiązaniem, bo chyba spojrzałeś tylko na kod bez pomyślenia. SENTRY_ENVIRONMENT domyślnie nie ma. Dopiero jeśli ktoś zechce nadpisać wartość kopiuje sobie config gdzie jest ta stała jest definiowana. Jak nie skopiuje to użyje wartości domyślnej.

Można oczywiście zrobić plik z domyślnymi wartościami, który będzie zawsze i includować go najpierw, ale to wtedy w nim trzeba dodatkowo sprawdzać czy dana stała już wcześniej nie została stworzona, aby móc to nadpisać w jeszcze kolejnym pliku. I trzeba jeszcze kolejnego pliku z przykładem jak je nadpisać. Stąd aby to uprościć poszedłem w ten sposób, można zrobić w inny, zasugeruj jaki ;)

Reply:
qa-config-example.php ma w sobie "define('SENTRY_ENVIRONMENT', 'local');". Brzmi jakby miał z domysłu wartość.

Reply:
Jeszcze raz: spójrz na całe rozwiązanie. Plik qa-config-example.php (w repo) nie jest w ogóle czytany przez Q2A, szuka tylko pliku qa-config.php (ignorowany przez Gita). Więc SENTRY_ENVIRONMENT nie ma z domysłu wartości, bo nie ma miejsca gdzie byłaby definiowana. Jeśli ktoś skopiuje qa-config-example do qa-config to owszem, będzie miał i tam może tę wartość zmienić.

Można zrobić wymóg istnienia qa-config.php, ale nie robiłem tego, bo wtedy trzeba zmienić też konfigurację vagranta, aby ten robił to automatycznie podczas stawiania środowiska

Reply:
> Można zrobić wymóg istnienia qa-config.php, ale nie robiłem tego, bo wtedy trzeba zmienić też konfigurację vagranta

I to mi chodziło po głowie własnie - domyślnie zwykle jest tak, że nie trzeba się martwić o takie rzeczy. Na chwilę obecną jest do zaakceptowania. Jednak wole wyjście by vagrant kopiowal sobie ten konfig z domysłu.

Reply:
W mojej propozycji też nie trzeba się martwić, bo są wartości domyślne i od razu działa. Rozumiem jednak jaka idea za tym idzie i ok, wartości domyślne w qa-base wtedy nie będą potrzebne, więc będzie powiedzmy nieco bardziej uporządkowane. Myślę że na to żeby vagrant sam skopiował możemy zmienić już teraz, nie jest to na tyle pilne, aby mergować byle było